### PR TITLE
Let tests access setup teardown container

### PR DESCRIPTION
**Pull request by clnoll**
To be able to test CDC DBs, DB Sources will need to be able to send commands to the database during tests where >1 sync is performed.

This PR gives the ability to do that by creating a fixture for the `setup_container` which is provided to the incremental tests.

DB Sources can then configure a `between_syncs_command` in their `acceptance-test-config.yml` that will run between syncs, in tests where more than one sync is performed.

### CODE REVIEW

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_incremental.py [hunk 1]
```diff
@@ -172,6 +177,10 @@ async def test_two_sequential_reads(
             state_input = states_1[-1].state.data
 
         # READ #2
+        if setup_container and setup_teardown_dockerfile_config.between_syncs_command:
+            detailed_logger.info(
+                await setup_container.with_exec(setup_teardown_dockerfile_config.between_syncs_command, skip_entrypoint=True).stdout()
```
Comment (alafanechere) [inline airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_incremental.py:182]: I'm wondering if we'd have to introduce a cache buster with an env var here to make sure the "between_syncs_command" is always run.

**Review by alafanechere — APPROVED** [review]
This "between sync command" does not sound utterly wrong to me, but this might mean we're starting to make the CAT interface evolve to specifically match DB needs.
I'm wondering if we could build a very generic setup/teardown flow in which what's executed can be dynamically determined by the setup/teardown container according to some context we could pass as env var.
In practice this would mean:
* We always run set/up and teardown before running any command on the container
* The set/up and teardown container is passed something like a state (test node id, test number etc.)
* According to this state / context the developers can build dynamic scripts bundled in the dockerfile.

I'm dropping this suggestion now, but I don't mean it to be blocking if this new "in between read" command is the last kind of hook you need to support the DB team usecase.

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_incremental.py (hunk 1 above)
Comment (clnoll) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_incremental.py:182]: The concern is that `with_exec` won't be run if it's already been run with the same command? The cache buster does seem like a good idea if that's the concern. That would look something like this?

```
await setup_container.with_env_variable("CACHEBUSTER", str(uuid4())).with_exec(setup_teardown_dockerfile_config.between_syncs_command, skip_entrypoint=True).stdout()
```

**Comment by clnoll** [comment]
@alafanechere that does seem like an interesting suggestion. This is the last hook that I can think of, so I'm going to punt on making that change but if the logic gets any more complicated I will circle back.

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_incremental.py (hunk 1 above)
Comment (alafanechere) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_incremental.py]: Yep exactly!

**Comment by clnoll** [comment]
@xiaohansong @theyueli please note the variable name changes here.

Your `acceptance-test-config.yml` should now have the following fields, for tests that require setup/teardown/etc.
```
        client_container_config:
          client_container_dockerfile_path: str  # path to your Dockerfile; we will mount secrets & the connector code into the container when we build it
          setup_command: List[str]  # command for setting up e.g. seeding the database
          teardown_command: List[str]  # command for any teardown needed
          between_syncs_command: Optional[List[str]]  # command for running between syncs that occur within the same test, e.g. to reseed the database
```

**Pull request merged**